api/admin/controller.py

```diff
@@ -1942,3 +1942,67 @@ def library_registrations(self, do_get=HTTP.get_with_timeout, do_post=HTTP.post_
 
         return Response(unicode(_("Success")), 200)
 
+    def sitewide_registration(self, do_get=HTTP.get_with_timeout, do_post=HTTP.post_with_timeout, key=None):
+        integration_id = flask.request.form.get("integration_id")
+
+        integration = get_one(self._db, ExternalIntegration, id=integration_id)
+        if not integration:
+            return MISSING_SERVICE
+
+        # Get the catalog for this service.
+        try:
+            response = do_get(integration.url, allowed_response_codes=['2xx', '3xx'])
+        except Exception as e:
+            return REMOTE_INTEGRATION_FAILED
+
+        catalog = response.json()
+        links = catalog.get('links', [])
+
+        # Get the link for registration from the catalog.
+        register_urls = filter(lambda l: l.get('rel')=='register', links)
```

Comment:
I would also check the `type` of the link to make sure it's not offering to register with a library registry or something. Should be looking for `application/opds+json;profile=https://librarysimplified.org/rel/profile/metadata-service`.